pyanaconda/modules/payloads/payload/rpm_ostree/installation.py

```diff
@@ -580,7 +599,11 @@ def run(self):
         # We use the UNTRUSTED flag as a way to help verification for potentially
         # corrupted ISOs - this way ostree will validate the checksum of objects as
         # it writes data.
-        pull_opts = {'refs': Variant('as', [ref]), 'flags': Variant('i', OSTree.RepoPullFlags.UNTRUSTED)}
+        if self._is_container:
+            pull_opts = {'flags': Variant('i', OSTree.RepoPullFlags.UNTRUSTED)}
```

Comment:
I don't think this is going to work.  It needs to instead fork off `ostree container image pull` if that's the goal (to separate pulling from deployment).

(Which i think may complicate things as that isn't part of the C library today exposed via pygobject, so progress reporting is more complex)

Reply:
I wanted to know more about this. I honestly not sure why we are pulling the remote then remove remote and do the deployment. What is the difference between deployment and pull?

Honestly, I don't think I care if it's done by one command or two. Less is better for us probably?

Reply:
The main reason to distinguish between the pulling and deployment is that pulling from the network is something for which one commonly wants e.g. a progress bar in a GUI environment.

The deployment phase is usually fast enough that it doesn't need it.

But indeed, today in Anaconda we could just skip this separate pull phase and rely on the deploy code automatically doing a pull if the image isn't present.

This whole discussion is really exactly the semantics of e.g.:

```
$ podman run busybox echo hello
```

If you don't have the `busybox` image present, podman will pull it by default.  But it also means that if the image pull fails (due to network flakes) the command will fail.  And so in many environments there's going to be a separation between pulling (and retries) versus the run/deployment.

Reply:
If the deployment is enough for this task we can probably just read command output to get progress reporting? Or for first version don't complicate things and instead just show "deploying". Could be improved later.

Should we leave the pull for ostree repository and bypass that just for container or there is no benefit in doing that?

Reply:
Yeah, worst case anaconda in the GUI could display a terminal with the status.

(That said, what will be glaring in this model is the fact that the ostree stack today is not translated, while Anaconda last I looked is...)